lib/api/objectCopy.js

```diff
@@ -65,6 +71,17 @@ function _prepMetadata(sourceObjMD, headers, sourceIsDestination, authInfo,
         utils.getMetaHeaders(sourceObjMD) :
         utils.getMetaHeaders(headers);
 
+    // If tagging directive is REPLACE, but you don't specify the
+    // x-amz-tagging in the request, the destination object will
+    // not have any tags.
```

Comment:
No but I can also add: `if tagging directive is COPY but the source object does not have tags, the destination object will not have any tags.`

There is no difference between saving {} and undefined both are considered as 0 tag.
`x-amz-tagging-count` header is returned only if the count is greater than zero.